mathesar/api/db/viewsets/tables.py

```diff
@@ -208,6 +209,46 @@ def existing_import(self, request, pk=None):
             temp_table.insert_records_to_existing_table(
                 target_table, data_files, mappings
             )
+        except NotNullError as e:
+            raise database_api_exceptions.NotNullImportViolationAPIException(
+                e,
+                message='Null values cannot be inserted into this column',
+                status_code=status.HTTP_400_BAD_REQUEST
+            )
+        except ForeignKeyError as e:
+            raise database_api_exceptions.ForeignKeyViolationAPIException(
+                e,
+                message='Cannot add an invalid reference to a record',
+                status_code=status.HTTP_400_BAD_REQUEST
+            )
+        except TypeMismatchError as e:
+            raise database_api_exceptions.TypeMismatchViolationAPIException(
+                e,
+                message='Type mismatch error',
+                status_code=status.HTTP_400_BAD_REQUEST
+            )
+        except UniqueValueError as e:
+            raise database_api_exceptions.UniqueImportViolationAPIException(
+                e,
+                message='This column has uniqueness constraint set so non-unique values cannot be inserted',
+                status_code=status.HTTP_400_BAD_REQUEST
+            )
+        except ExclusionError as e:
+            raise database_api_exceptions.ExclusionViolationAPIException(
+                e,
+                message='This record violates exclusion constraint',
+                status_code=status.HTTP_400_BAD_REQUEST
+            )
+        except IntegrityError as e:
+            raise database_base_api_exceptions.IntegrityAPIException(
+                e,
+                status_code=status.HTTP_400_BAD_REQUEST
+            )
+        except ProgrammingError as e:
+            raise database_base_api_exceptions.ProgrammingAPIException(
+                e,
+                status_code=status.HTTP_400_BAD_REQUEST
+            )
         except Exception as e:
             # ToDo raise specific exceptions.
             raise e
```

Comment:
This is redundant now. The error will be raised if the exception doesn't match any `except` block above.

Reply:
Please refer to my comment above.